argilla-server/src/argilla_server/validators/datasets.py

```diff
@@ -22,16 +22,27 @@
 
 class DatasetCreateValidator:
     @classmethod
-    async def validate(cls, db, dataset: Dataset) -> None:
+    async def validate(cls, db: AsyncSession, dataset: Dataset) -> None:
         await cls._validate_workspace_is_present(db, dataset.workspace_id)
         await cls._validate_name_is_not_duplicated(db, dataset.name, dataset.workspace_id)
 
     @classmethod
-    async def _validate_workspace_is_present(cls, db, workspace_id: UUID) -> None:
+    async def _validate_workspace_is_present(cls, db: AsyncSession, workspace_id: UUID) -> None:
         if await Workspace.get(db, workspace_id) is None:
             raise UnprocessableEntityError(f"Workspace with id `{workspace_id}` not found")
 
     @classmethod
-    async def _validate_name_is_not_duplicated(cls, db, name: str, workspace_id: UUID) -> None:
+    async def _validate_name_is_not_duplicated(cls, db: AsyncSession, name: str, workspace_id: UUID) -> None:
         if await Dataset.get_by(db, name=name, workspace_id=workspace_id):
             raise NotUniqueError(f"Dataset with name `{name}` already exists for workspace with id `{workspace_id}`")
+
+
+class DatasetUpdateValidator:
+    @classmethod
+    async def validate(cls, db: AsyncSession, dataset: Dataset, dataset_attrs: dict) -> None:
```

Comment:
Even if I don't need the class method to be asynchronous, and not needing the session I prefer to have the same signature in all of them.